server/src/main/java/com/broadcom/lsp/domain/cobol/databus/impl/CopybookRepositoryLRU.java

```diff
@@ -40,111 +39,55 @@
 @Singleton
 public class CopybookRepositoryLRU implements CopybookRepository {
   @Getter private final int cacheMaxSize;
-  @Setter @NonNull private ArrayList<CopybookStorable> cpyRepo;
-  private final Comparator<CopybookStorable> storableComparator =
-      Comparator.comparingInt(CopybookStorable::getHit); // Time last recently used
-  private final Comparator<CopybookStorable> timeComparator =
-      Comparator.comparingLong(CopybookStorable::getGenDt);
-
-  @NonNull private final AtomicBoolean isSort = new AtomicBoolean(false);
+  private Cache<String, CopybookModel> cache;
 
   @Inject
   public CopybookRepositoryLRU(@Named("CACHE-MAX-SIZE") int cacheSize) {
     cacheMaxSize = cacheSize;
-    cpyRepo = new ArrayList<>(cacheMaxSize);
-  }
-
-  @Override
-  public synchronized void sortCache() {
-    if (!isSort.get()) {
-      cpyRepo.sort(storableComparator.reversed().thenComparing(timeComparator));
-    }
-    isSort.set(true);
-  }
-
-  @Override
-  public Optional<CopybookStorable> getCopybookStorableFromCache(@NonNull long uuid) {
-    ArrayList<CopybookStorable> shallowCpy = (ArrayList<CopybookStorable>) cpyRepo.clone();
-    Optional<CopybookStorable> cpy =
-        shallowCpy.stream().filter(copy -> uuid == copy.getId()).findAny();
-    if (cpy.isPresent()) return cpy.map(SerializationUtils::clone);
-    return cpy;
+    cache = CacheBuilder.newBuilder()
+        .maximumSize(cacheMaxSize)
+        .expireAfterAccess(3, TimeUnit.HOURS)
```

Comment:
I rewrite that has been. In `CopybookStorable` we had this:
```
  private static final long TTU = 3600000L * 3L;
```
This means hardcoded 3 hours after writing to the cache. I've decided to change it to 3 hours after the last access.
I agree that we could make this value configurable. Do you think that we need this configurable value in hours or minutes? In case when retention policy will be half an hour.

Reply:
I think that 3 hours after writing is ok. We may either provide the time unit as a configuration or use hardcoded seconds or milliseconds and provide only the value. 
Due to it is not something that we are going to change rarely, I would prefer the second option.

Reply:
Since we're planning here to do refactor here I think could be nice to move those config in a property file in order to have a better customization for future needs

Reply:
All right. I will make duration and time unit configurable but initialize them as we do for `cache-max-size` already.
I agree, that having a separate property file will be a good option. I'll do it in a separate PR.